src/main/java/seedu/address/model/person/Person.java

```diff
@@ -35,7 +35,7 @@ public Person(Name name, Email email, Gender gender, Age age, Set<Tag> tags, Det
         this.email = email;
         this.gender = gender;
         this.age = age;
-        this.detail = detail;
+        this.detail = (detail != null) ? detail : new Detail("");
```

Comment:
Do you need the string argument for the Detail constructor? In `Detail.java`, you added a new default constructor (line 23) so using the default constructor is sufficient right?

Reply:
Use this.detail = detail seems to fail some test cases, I guess I'll keep it like this for now.

Reply:
Sorry, I think I wasn't clear in my comments. I meant `this.detail = (detail != null) ? detail : new Detail();` instead of `new Detail("")`.